diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e49c19e..552ad462 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -691,9 +691,10 @@ write_basic_package_version_file("${Ceres_BINARY_DIR}/CeresConfigVersion.cmake"
 # when read in by a client project as part of find_package(Ceres) creates
 # imported library targets for Ceres (with dependency relations) which can be
 # used in target_link_libraries() calls in the client project to use Ceres.
-install(EXPORT CeresExport
-        NAMESPACE Ceres::
-        DESTINATION ${RELATIVE_CMAKECONFIG_INSTALL_DIR} FILE CeresTargets.cmake)
+# When using FetchContent with both Eigen and Ceres, the next line causes an error during CMake config
+#install(EXPORT CeresExport
+#        NAMESPACE Ceres::
+#        DESTINATION ${RELATIVE_CMAKECONFIG_INSTALL_DIR} FILE CeresTargets.cmake)
 
 # Save the relative path from the installed CeresConfig.cmake file to the
 # install prefix.  We do not save an absolute path in case the installed package
diff --git a/internal/ceres/file.cc b/internal/ceres/file.cc
index 60d35fac..5ee76580 100644
--- a/internal/ceres/file.cc
+++ b/internal/ceres/file.cc
@@ -34,6 +34,7 @@
 
 #include <cstdio>
 #include <string>
+#include <cstdint>
 
 #include "glog/logging.h"
 
