diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6fbb37..311a7e3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.4.4)
+cmake_minimum_required(VERSION 3.15)
 set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS ON)
 
 project(zlib C)
@@ -226,6 +226,10 @@ if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL )
     install(FILES ${ZLIB_PC} DESTINATION "${INSTALL_PKGCONFIG_DIR}")
 endif()
 
+
+export( TARGETS zlib NAMESPACE ZLIB:: FILE zlibConfig.cmake)
+
+
 #============================================================================
 # Example binaries
 #============================================================================
