--- /Users/wcjohns/install/wt-3.3.1/src/Wt/WCanvasPaintDevice.C	2014-08-29 13:34:56.000000000 -0700
+++ /Users/wcjohns/Downloads/WCanvasPaintDevice.C	2014-08-29 13:36:29.000000000 -0700
@@ -20,6 +20,10 @@
 #include <sstream>
 #include <boost/lexical_cast.hpp>
 
+#if __APPLE__
+#include "TargetConditionals.h"
+#endif
+
 namespace {
   static const double EPSILON = 1E-5;
 }
@@ -112,6 +116,12 @@
 	textMethod_ = Html5Text;
     }
   }
+  
+#if __APPLE__
+#if( defined(TARGET_IPHONE_SIMULATOR) || defined(TARGET_OS_IPHONE) )
+  textMethod_ = Html5Text;
+#endif
+#endif
 }
 
 WFlags<WPaintDevice::FeatureFlag> WCanvasPaintDevice::features() const
