--- wt-3.3.4/src/Wt/Dbo/ptr_impl.h.orig	2020-11-15 12:53:36.000000000 -0800
+++ wt-3.3.4/src/Wt/Dbo/ptr_impl.h	2020-11-15 12:59:24.000000000 -0800
@@ -46,7 +46,7 @@
     }
 
 template <class C>
-MetaDbo<C>::~MetaDbo()
+MetaDbo<C>::~MetaDbo()  noexcept(false)
 {
   if (refCount_)
     throw std::logic_error("Dbo: refCount > 0");
@@ -302,7 +302,7 @@
 }
 
 template <class C>
-ptr<C>::~ptr()
+ptr<C>::~ptr() noexcept(false)
 {
   freeObj();
 }
